src/interpreter_lifecycle.rs

```diff
@@ -0,0 +1,140 @@
+use std::sync;
+
+use crate::{ffi, internal::state::AttachGuard, Python};
+
+static START: sync::Once = sync::Once::new();
+
+/// Prepares the use of Python in a free-threaded context.
+///
+/// If the Python interpreter is not already initialized, this function will initialize it with
+/// signal handling disabled (Python will not raise the `KeyboardInterrupt` exception). Python
+/// signal handling depends on the notion of a 'main thread', which must be the thread that
+/// initializes the Python interpreter.
+///
+/// If the Python interpreter is already initialized, this function has no effect.
+///
+/// This function is unavailable under PyPy because PyPy cannot be embedded in Rust (or any other
+/// software). Support for this is tracked on the
+/// [PyPy issue tracker](https://github.com/pypy/pypy/issues/3836).
+///
+/// # Examples
+/// ```rust
+/// use pyo3::prelude::*;
+///
+/// # fn main() -> PyResult<()> {
+/// pyo3::prepare_freethreaded_python();
+/// Python::attach(|py| py.run(pyo3::ffi::c_str!("print('Hello World')"), None, None))
+/// # }
+/// ```
+pub fn prepare_freethreaded_python() {
```

Comment:
Should we take this opportunity to rename this to something more descriptive? Maybe `initialize_python` or even `Python::initialize`?